packages/pipeline/src/parsers/copper/index.ts

```diff
@@ -0,0 +1,274 @@
+}
+export interface CopperActivityParentResponse {
+    id: number;
+    type: string;
+}
+
+// custom activity types
+export enum CopperActivityTypeCategory {
+    user = 'user',
+    system = 'system',
+}
+export interface CopperActivityTypeResponse {
+    id: number;
+    category: CopperActivityTypeCategory;
+    name: string;
+    is_disabled?: boolean;
+    count_as_interaction?: boolean;
+}
+
+export interface CopperOpportunityResponse {
+    id: number;
+    name: string;
+    assignee_id?: number;
+    close_date?: string;
+    company_id?: number;
+    company_name?: string;
+    customer_source_id?: number;
+    loss_reason_id?: number;
+    pipeline_id: number;
+    pipeline_stage_id: number;
+    primary_contact_id?: number;
+    priority?: string;
+    status: string;
+    tags: string[];
+    interaction_count: number;
+    monetary_value?: number;
+    win_probability?: number;
+    date_created: number; // in seconds
+    date_modified: number; // in seconds
+    custom_fields: CopperNestedCustomFieldResponse[];
+}
+interface CopperNestedCustomFieldResponse {
+    custom_field_definition_id: number;
+    value?: string;
+}
+// custom fields
+export enum CopperCustomFieldType {
+    String = 'String',
+    Text = 'Text',
+    Dropdown = 'Dropdown',
+    MultiSelect = 'MultiSelect', // not in API documentation but shows up in results
+    Date = 'Date',
+    Checkbox = 'Checkbox',
+    Float = 'Float',
+    URL = 'URL',
+    Percentage = 'Percentage',
+    Currency = 'Currency',
+    Connect = 'Connect',
+}
+export interface CopperCustomFieldOptionResponse {
+    id: number;
+    name: string;
+}
+export interface CopperCustomFieldResponse {
+    id: number;
+    name: string;
+    data_type: CopperCustomFieldType;
+    options?: CopperCustomFieldOptionResponse[];
+}
+/**
+ * Parse response from Copper API /search/leads/
+ *
+ * @param leads - The array of leads returned from the API
+ * @returns Returns an array of Copper Lead entities
+ */
+export function parseLeads(leads: CopperLeadResponse[]): CopperLead[] {
+    return leads.map(lead => {
+        const entity = new CopperLead();
+        entity.id = lead.id;
+        entity.name = lead.name;
+        entity.firstName = lead.first_name;
+        entity.lastName = lead.last_name;
+        entity.middleName = lead.middle_name;
+        entity.assigneeId = lead.assignee_id;
+        entity.companyName = lead.company_name;
+        entity.customerSourceId = lead.customer_source_id;
+        entity.monetaryValue = lead.monetary_value;
+        entity.status = lead.status;
+        entity.statusId = lead.status_id;
+        entity.title = lead.title;
+
+        entity.dateCreated = lead.date_created * ONE_SECOND;
+        entity.dateModified = lead.date_modified * ONE_SECOND;
+
+        entity.email = lead.email === null ? 0 : 1;
+
+        // nullable fields
+        entity.socials = 0;
+        entity.websites = 0;
+        entity.phoneNumbers = 0;
+        try {
```

Comment:
It looks like this would result in all of the `socials`, `websites`, and `phoneNumbers` fields being `0` if just one of them is `undefined` in the raw response from the API. That doesn't seem right.

But I have another question that might override this. Is it really all that useful to store *just* the number of `socials`, `websites`, and `phoneNumbers`? It seems like we should either be storing the values themselves (e.g., save each phone number) or nothing.

Reply:
> It looks like this would result in all of the socials, websites, and phoneNumbers fields being 0 if just one of them is undefined in the raw response from the API. That doesn't seem right.

Yeah, you are right. Thanks for catching that!

> But I have another question that might override this. Is it really all that useful to store just the number of socials, websites, and phoneNumbers? It seems like we should either be storing the values themselves (e.g., save each phone number) or nothing.

I don't think we should store these details because we would then be subject to GDPR requirements or potential future data protection regulation, and generally it's a privacy and data management concern. If we need this info for business purposes it lives within Copper - we shouldn't need it for analytics. 

I'm cool with not storing it if the numbers aren't of interest to us.

Reply:
Checked with Clay, he says numbers aren't useful.